compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -429,14 +429,14 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                             return;
                         }
                         let trait_ref = trait_predicate.to_poly_trait_ref();
-
-                        let (post_message, pre_message, type_def) = self
+                        let (self_ty, _) = self.tcx.short_ty_string(trait_ref.skip_binder().self_ty());
+                                                let (post_message, pre_message, type_def) = self
                             .get_parent_trait_ref(obligation.cause.code())
-                            .map(|(t, s)| {
+                            .map(|(_, s)| {
                                 (
-                                    format!(" in `{t}`"),
-                                    format!("within `{t}`, "),
-                                    s.map(|s| (format!("within this `{t}`"), s)),
+                                    format!(" in `{self_ty}`"),
+                                    format!("within `{self_ty}`, "),
+                                    s.map(|s| (format!("within this `{self_ty}`"), s)),
```

Comment:
you also need to print the shortening somewhere like
https://github.com/rust-lang/rust/blob/94bc9c737ea97d56f9b4076553ac37c05c1e3931/compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs#L3053

though I gotta say, it feels a bit ad-hoc and Prime to errors to have to make all these call sites do this manually.. do you have any better ideas @estebank?

Reply:
i presumed that was already done as I'm not modifying where it is being printed, only the message?

Reply:
I'm not sure, maybe you're right. You should add a test to see :)

Reply:
I agree that it is error prone to force every error that does this add the note, but on the other hand there are really only a handful of places where type shortening is needed in practice. Don't think we should force that to be addressed in this PR.

Reply:
i have just checked the variables returned from the tuple, and they are all used in some way or another